apps/aeprimop/src/aeprimop.erl

```diff
@@ -1122,17 +1164,20 @@ ga_attach({OwnerPubkey, GasLimit, GasPrice, ABIVersion,
            VMVersion, SerializedCode, AuthFun, CallData, Fee, Nonce}, S) ->
     assert_ga_active(S),
     RollbackS = S,
-    TotalAmount    = Fee + GasLimit * GasPrice,
-    {Account, S1}  = get_account(OwnerPubkey, S),
+    TotalAmount       = Fee + GasLimit * GasPrice,
+    {OwnerAmount, S0} = split_payment(TotalAmount, 0, S),
+    {Account, S1}     = get_account(OwnerPubkey, S0),
     assert_basic_account(Account), %% No re-attach
-    assert_account_balance(Account, TotalAmount),
+    assert_account_balance(Account, OwnerAmount),
     assert_ga_create_version(ABIVersion, VMVersion, S),
     Code           = assert_ga_attach_byte_code(ABIVersion, SerializedCode,
                                                 CallData, AuthFun, S),
     %% Charge the fee, the gas (the unused portion will be refunded)
     %% and the deposit (stored in the contract) to the contract owner (caller),
     CTVersion      = #{vm => VMVersion, abi => ABIVersion},
-    S2             = account_spend(Account, TotalAmount, S1),
+    S2             = if OwnerAmount > 0 -> account_spend(Account, OwnerAmount, S1);
```

Comment:
I have a hard time to see when the OwnerAmount == 0 kicks in? Let alone <0.

Reply:
OwnerAmount can't be `< 0` and if it is `0` the spending `0` is a no-op so we skip that.